test/functional/test_framework/test_node.py

```diff
@@ -141,6 +160,33 @@ def node_encrypt_wallet(self, passphrase):
         self.rpc = None
         self.rpc_connected = False
 
+    def add_p2p_connection(self, p2p_conn_type, dstaddr='127.0.0.1', dstport=None, services=NODE_NETWORK, send_version=True):
+        """Add a p2p connection to the node."""
+        if dstport is None:
+            dstport = p2p_port(self.index)
+        p2p_conn = p2p_conn_type()
+        self.p2ps.append(p2p_conn)
+        p2p_conn.add_connection(NodeConn(dstaddr, dstport, self.rpc, p2p_conn, services=services, send_version=send_version))
+        if self.p2p is None:
+            self.p2p = self.p2ps[0]
+        self.p2p_connected = True
+        self.p2p_ever_connected = True
+
+    def send_message(self, message):
+        """Send a p2p message to the node."""
+        if not self.p2p_ever_connected:
+            self.add_p2p_connections()
```

Comment:
Where is `add_p2p_connections` defined? Is it virtual?

Reply:
oops. This is vestigial. I previously had an `add_p2p_connections()`, but it turned out to not be very useful. I'll fix this up. Thanks for pointing this out.